modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc

```diff
@@ -0,0 +1,262 @@
+
+== Use Your Types in a DataWeave Script
+
+Having this directive in the script header loads all the existing types in the JSON schema. In this case, the only existing type is the *Root* type, specified at the root which describes an Object that can have three properties (`firstName`, `lastName`, and `age`). It is equivalent to declaring the following type in your DataWeave script:
+
+.DataWeave Script
+[source,dataweave,linenums]
+----
+%dw 2.0
+type Root = { firstName: String, lastName: String, age?: Number }
+----
+
+Notice that the age field is the only one that is optional and it is signaled by the `?`.
+
+This is how you can use the type to know if a value follows the structure defined by the JSON schema.
+The following example outputs the value `true` since the object contains the required fields, `firstName` and `lastName`:
+
+.DataWeave Script
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from jsonschema!jsonschema!example::schema::Person
+---
+{
+ firstName: "John",
+ lastName: "Doe"
+} is Root
+----
+
+.Output
+----
+"True"
+----
+
+The following example outputs the value `false` since the object does not contain the required fields, `firstName` and `lastName`:
+
+.DataWeave Script
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from jsonschema!jsonschema!example::schema::Person
+---
+{
+ firstName: "John",
+ age: 20
+} is Root
+----
+
+.Output
+----
+"False"
+----
+
+== Use Definitions Inside Schemas
+
+Definitions inside JSON schemas are also loaded when the module is imported:
+
+.JSON Script
+[source,JSON]
+----
+{
+ "$schema": "http://json-schema.org/draft-04/schema#",
+ "definitions": {
+   "address": {
+     "type": "object",
+     "properties": {
+       "street_address": {
+         "type": "string"
+       },
+       "city": {
+         "type": "string"
+       },
+       "state": {
+         "type": "string"
+       }
+     },
+     "required": [
+       "street_address",
+       "city",
+       "state"
+     ]
+   },
+   "person": {
+     "type": "object",
+     "properties": {
+       "firstName": {
+         "type": "string",
+         "description": "The person's first name."
+       },
+       "lastName": {
+         "type": "string",
+         "description": "The person's last name."
+       },
+       "age": {
+         "description": "Age in years which must be equal to or greater than zero.",
+         "type": "integer",
+         "minimum": 0
+       }
+     },
+     "required": [
+       "firstName",
+       "lastName"
+     ]
+   }
+ },
+ "type": "object",
+ "properties": {
+   "person": {
+     "$ref": "#/definitions/person"
+   },
+   "address": {
+     "$ref": "#/definitions/address"
+   }
+ }
+}
+----
+
+You can import the types from that schema with the following directive:
```

Comment:
Can we say something like we're assuming the path to the schema above is example/schema/Account.json or something so that we could use that directive to import all the types from it?

Reply:
```suggestion
You can import the types from that schema with the following directive, assuming the JSON schema is `example/schema/Account.json`:
```

Reply:
yes, that would be clearer.

Reply:
Modification added.